### DISCUSSION TITLE
Is zrange with bylex supported?

### DISCUSSION THREAD

**Discussion by kchonka** [dsoftwareinc/fakeredis-py#122] [Q&A] [answered]
Not sure if it's a bug yet, but I'm getting a `redis.exceptions.ResponseError: min or max is not a float` error when I try to use zrange with the bylex=True arg

**Comment by cunla** [comment, marked as answer]
Hi, I see you opened a bug for this. Would you like to fix it as you pretty much pinpointed the issue (`zrange` doesn't handle `bylex` param)?

**Comment by kchonka** [comment]
Yep, can do!

**Reply by cunla** [reply]
Great, thank you.

Some pointers:
- There is a version variable coming from one of the mixin you can use, so you can add a condition [here](https://github.com/cunla/fakeredis-py/blob/master/fakeredis/commands_mixins/sortedset_mixin.py#L161-L169) for `bylex` and version (something like `casematch(arg, b'byscore') and self.version>=7`).
- Alternatively, you can scan the args in [`SortedSetCommandsMixin.zrange`](https://github.com/cunla/fakeredis-py/blob/master/fakeredis/commands_mixins/sortedset_mixin.py#L182) and call `_zrangebylex` if `bylex` is on.
- When writing tests for it, you can use `@pytest.mark.min_server(...)` and `@pytest.mark.max_server(...)`.
Please write tests to see that on redis6 `bylex` is raising an error and for redis7 `bylex` is behaving as expected.

I am happy to have another person digging into the small details which I might have missed. If you are interested in being more involved - please sign up [here](https://www.codetriage.com/cunla/fakeredis-py). After a few PRs, I will add you to the repo collaborators.
